core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1992,4 +1992,32 @@ package object config {
       .version("3.1.0")
       .doubleConf
       .createWithDefault(5)
+
+  private[spark] val SHUFFLE_NUM_PUSH_THREADS =
+    ConfigBuilder("spark.shuffle.push.numPushThreads")
+      .doc("Specify the number of threads in the block pusher pool. These threads assist " +
+        "in creating connections and pushing blocks to remote shuffle services when push based " +
+        "shuffle is enabled. By default, the threadpool size is equal to the number of cores.")
+      .version("3.1.0")
+      .intConf
+      .createOptional
+
+  private[spark] val SHUFFLE_MAX_BLOCK_SIZE_TO_PUSH =
+    ConfigBuilder("spark.shuffle.push.maxBlockSizeToPush")
+      .doc("The max size of an individual block to push to the remote shuffle services when push " +
+        "based shuffle is enabled. Blocks larger than this threshold are not pushed.")
```

Comment:
BTW, `Blocks larger than this threshold are not pushed.` might be a big misleading point for the users. Could you add a little more description about what happens for those big non-pushed shuffle blocks? It helps the user feel safer.

Reply:
done